apps/src/codebridge/FileBrowser/FileBrowserHeaderPopUpButton.tsx

```diff
@@ -43,40 +44,47 @@ export const FileBrowserHeaderPopUpButton = () => {
   return (
     <>
       <FileUploaderComponent />
-      <PopUpButton iconName="plus" alignment="left" id="uitest-files-plus">
-        <PopUpButtonOption
-          iconName="plus"
-          labelText={codebridgeI18n.newFolder()}
-          clickHandler={() =>
-            openNewFolderPrompt({parentId: DEFAULT_FOLDER_ID})
-          }
-        />
-        <PopUpButtonOption
-          iconName="plus"
-          labelText={codebridgeI18n.newFile()}
-          clickHandler={() => openNewFilePrompt({folderId: DEFAULT_FOLDER_ID})}
-          id="uitest-new-file"
-        />
-
-        <PopUpButtonOption
-          iconName="upload"
-          labelText={codebridgeI18n.uploadFile()}
-          clickHandler={() => startFileUpload()}
-        />
-        {experiments.isEnabled(experiments.PYTHONLAB_BACKPACK) && (
+      {isLoading ? (
+        <i className="fa fa-spinner fa-spin" />
```

Comment:
I am wondering if it would make more sense to put the spinner in the modal instead. When we are loading files we can replace the list of files with a spinner rather than having the button turn into a spinner (when I first watched your video I didn't see the spinner).

Reply:
@breville has also advocated for fading in spinners after a delay so they don't flash on the screen. The lab2 loading screen is an example of this: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/lab2/views/loading.module.scss (we don't need to be as complicated here, but a similar idea could apply)

Reply:
> @breville has also advocated for fading in spinners after a delay so they don't flash on the screen. The lab2 loading screen is an example of this: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/lab2/views/loading.module.scss (we don't need to be as complicated here, but a similar idea could apply)

Thanks! Btw, I had some issues trying to add a style rule for the icon to the css module.  If you're good with the `PendingDialog` component, definitely open to iterating on the style.